pkgs/tools/admin/bom/default.nix

```
@@ -0,0 +1,67 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
+
+buildGoModule rec {
+  pname = "bom";
+  version = "0.4.1";
+
+  src = fetchFromGitHub {
+    owner = "kubernetes-sigs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-HgBpEIUaeVCMkbu1KtBh0LGZQGFLyj3rF0cYDt8vDRk=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD > $out/COMMIT
+      # '0000-00-00T00:00:00Z'
+      date -u -d "@$(git log -1 --pretty=%ct)" "+'%Y-%m-%dT%H:%M:%SZ'" > $out/SOURCE_DATE_EPOCH
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
+  };
+  vendorSha256 = lib.fakeSha256;
```

Why? O.o

---

It didn't worked on my machine so I don't know how to find that value? Can you tell me how can I calculate this sha256 without building a package? 😇

---

Usually you put this because after one or two iterations Nix will complain about failed hash and return the expected hash.

On other occasions, this value is unneeded-but-required. Then you should set it to null:

```suggestion
  vendorHash = null;
```

Indeed the docs (doc/language-frameworks/go.section.md) explain this special case.